osu.Game/Screens/Multi/Multiplayer.cs

```diff
@@ -269,6 +263,25 @@ public override bool OnExiting(IScreen next)
             return false;
         }
 
+        public override bool OnBackButton()
+        {
+            var songSelect = screenStack.CurrentScreen as MatchSongSelect;
+
+            if (songSelect?.ModSelect.State.Value == Visibility.Visible)
+            {
+                songSelect.ModSelect.Hide();
+                return true;
+            }
```

Comment:
not sure `Multiplayer` should be prodding its sub-screen states directly like this. maybe something an `OnBackButton()` in `IMultiplayerSubScreen` would be better.

Reply:
I think it needs to be done here, as sub screens can't control the back button.

```
if ((screenStack.CurrentScreen as IMultiplayerSubScreen).OnBackButton())
	return true;
```

I thought of this, seems more better?

Reply:
yeah, that's what I mean; the check itself being there I don't have much of an issue with, but accessing the subscreens' controls in here (some of which need to be exposed for this to work) doesn't seem too great.